boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -164,6 +164,10 @@ ifeq ($(CONFIG_LIBCXX),y)
   # It makes libcxx __availability header unhappy.
   # https://github.com/llvm/llvm-project/blob/2e2999cd44f6ec9a5e396fa0113497ea82582f69/libcxx/include/__availability#L258
   CXXFLAGS += -D_LIBCPP_DISABLE_AVAILABILITY
+
+  # Force inclusion of Nuttx math.h, which undefines __GLIBC__.
+  # Without this inclusing, the system math.h may be picked up first, casuing an incorrect configuration in llvm.
+  CXXFLAGS += -include$(TOPDIR)$(DELIM)include$(DELIM)arch$(DELIM)math.h 
```

Comment:
let add CONFIG_ARCH_MATH_H to the related defconfig instead

Reply:
I overlooked this option. Adding it to `defconfig` solves this issue.